### ISSUE TITLE
Markdown editing support

### ISSUE THREAD

**Issue by brenmcnamara** [PowerNote/Notebook#13]
- When editing the document and adding basic markdown-recognized syntax, should apply the relevant styling.
- Use dropbox paper as an example of how to implement the markdown features

### Bold / italicize

- You can surround text with single or double asterisks to bold or italicize it
- Some subtleties of how the feature works in dropbox paper
  - surrounding a text with asterisks (\*Hello\*), will italicize text after a space or newline is entered following the final asterisk
  - if you write text with asterisks in it using a non-markdown editor (\*Hello\* \*World\*), then paste it in, it will not style the text
  - if there is a space immediately after the first asterisk or immediately before the last asterisk, styling is not applied (\* Hello\* or \*Hello \*). If there are spaces in between the asterisks but not touching either of the asterisks, the styling is applied (\*Hello World\*)

### Links

- Markdown-style links are not supported in dropbox paper (\[text\]\(link\))
- Links are created when a valid url is added (or by using some of the manual methods for adding links)

### Headers

- When adding the hashtags at the beginning of a line, then pressing a space, the hashtags disappear and the subsequent text will become a header
- Hashtags followed by a newline are NOT styled (the hashtags remain)
- 1, 2, or 3 simultaneous hashtags are supposed (header 1, header 2, header 3), but no more than that
- hashtags in the middle of the line (not at the beginning) are treated as tags

### In a nutshell

* Words / phrases / patterns are committed (usually after whitespace). The committed patterns are then styled if they match a certain way.

**Comment by brenmcnamara** [comment]
As pointed out [here](https://github.com/facebook/draft-js/issues/506), we should use the [onBeforeHandleInput](https://draftjs.org/docs/api-reference-editor.html#handlebeforeinput) method to get the last character in the list.

**Comment by brenmcnamara** [comment]
Found [this](https://github.com/facebook/draft-js/issues/132), may be useful when implementing checklist feature
